course/lesson_04_unit_testing/README.md

```diff
@@ -1,20 +1,25 @@
 ### Lesson 4 - Unit testing
+Unit testing is very wide area of engineering. There are plenty of tools and frameworks supporing developers and testers in tests implementation and execution. In this lesson we'll go through 2 poular testing frameworks: **unittest** and **pytest**.
+
+**Please read the introduction materials carefully - especially if you've never implemented your own unit tests. Your main goal in this lesson is to catch the idea of unit testing.**
+
+#### Introduction materials
+
+- [What is Unit Testing?](https://code.tutsplus.com/articles/the-beginners-guide-to-unit-testing-what-is-unit-testing--wp-25728) (section *What is Unit Testing?* only)
 - [Testing basics](http://docs.python-guide.org/en/latest/writing/tests/) (focus on *general rules of testing*)
 - [Introduction to unittest](http://www.voidspace.org.uk/python/articles/introduction-to-unittest.shtml) (sections: *Introduction, Testing, The unittest Module, The assert methods*, skip the rest)
 - [py.test unit tests](https://docs.pytest.org/en/latest/getting-started.html#getstarted) better way to test (it's easier than you think)
 
 ### practice projects
 
-1. Using `unittest.TestCase` write 4 unit tests for `collatz()` function 
-(not the whole script, just the function):
+1. Using `unittest.TestCase` write 4 unit tests for `collatz()` function (not the whole script, just the function). Base on [25.3.1. Basic example](https://docs.python.org/2/library/unittest.html#basic-example). Don't focus too much on `class` keyword in test definition. Treat this as a code that must be there for tests to be ran. Implement following tests:
```

Comment:
How about link to python 3.6 docs ?

https://docs.python.org/3.6/library/unittest.html#basic-example

Reply:
done